### PR TITLE
Refactoring to support storage modifiers

### PR DESCRIPTION
**Pull request by folbricht**
This change abstract the handling of uncompressed chunk storage with the goal of supporting other storage modifier layers later, namely for encryption.

### CODE REVIEW

File: coverter.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
+package desync
```
Comment (charles-dyfis-net) [inline coverter.go:1]: Is there a typo in this filename?

**Review by charles-dyfis-net — APPROVED** [review]
This is elegant, well-tested and well-commented. If you were to merge this (as the groundwork for adding encryption) as a separate PR from the actual crypto implementation, I could see merging it as-is as a code-quality enhancement (albeit maybe retitled).

File: coverter.go (hunk 1 above)
Comment (folbricht) [reply coverter.go:1]: good catch. Thanks, Fixed

**Comment by folbricht** [comment]
I do actually have most of the remaining encryption code ready to go on top of this, with a few more modifications to the large refactoring (just missing a couple tests and docs). You're right, should probably use this PR for the refactoring only and add the encryption layer in another one.

**Pull request merged**